### DISCUSSION TITLE
PAT Basic 1025. 反转链表 (25) (C语言实现) | PAT Solution in C

### DISCUSSION THREAD

**Discussion by xlucn** [xlucn/PAT#74] [General]
https://oliverlew.github.io/PAT/Basic/1025.html 

 题目

**Comment by wacodespace** [comment]
node nodes[100000] = {0};
结构体可以这样初始化吗？

**Comment by xlucn** [comment]
> @Epimenides7
>  node nodes[100000] = {0};
> 结构体可以这样初始化吗？
> 

是可以的，类似于数组的初始化，这样做的结果就是结构数组所有成员都是0。

**Comment by wacodespace** [comment]
> @OliverLew
> > @Epimenides7
> >  node nodes[100000] = {0};
> > 结构体可以这样初始化吗？
> > 
> 
> 是可以的，类似于数组的初始化，这样做的结果就是结构数组所有成员都是0。

谢谢回复！

**Comment by wacodespace** [comment]
你好，我看了好长时间代码，还是不太理解
  if(np[j]->addr == (i ? np[i - 1]->next : A))
的含义，可以解释一下吗，万分感谢

**Comment by xlucn** [comment]
> @Epimenides7
> 你好，我看了好长时间代码，还是不太理解
>   if(np[j]->addr == (i ? np[i - 1]->next : A))
> 的含义，可以解释一下吗，万分感谢
> 

没问题。其实我写的可能逻辑不是很明晰，我最终的目标是**将链表顺序放于数组中**。因此涉及一个分类讨论：
- 数组第一个`i=0`地址应该对应首个结点的地址（代码里的`A`），
- 数组中第`i>0`个地址`np[i]->addr`应为前一个所指向的地址`np[i - 1]->next`

**Comment by wacodespace** [comment]
> @OliverLew
> > @Epimenides7
> > 你好，我看了好长时间代码，还是不太理解
> >   if(np[j]->addr == (i ? np[i - 1]->next : A))
> > 的含义，可以解释一下吗，万分感谢
> > 
> 
> 没问题。其实我写的可能逻辑不是很明晰，我最终的目标是**将链表顺序放于数组中**。因此涉及一个分类讨论：
> - 数组第一个`i=0`地址应该对应首个结点的地址（代码里的`A`），
> - 数组中第`i>0`个地址`np[i]->addr`应为前一个所指向的地址`np[i - 1]->next`

终于理解了，之前也一直在猜Node np[100000]这个数组的作用，听了您的解释终于明白了是用来放顺序链表的，感觉这个程序很精妙。
